src/types.ts

```diff
@@ -110,10 +115,24 @@ export const aliasSchema = z.string().regex(/^[\da-z-]+$/);
 
 export type Alias = z.infer<typeof aliasSchema>;
 
-export const chainAlias = aliasSchema.refine(
-  (value) => CHAINS.some((chain) => chain.alias === value),
-  (value) => ({ message: `Invalid chain alias: ${value}` })
-);
+export const chainAlias = z.string().superRefine((value, ctx) => {
+  if (!/^[\da-z-]+$/.test(value)) {
+    ctx.addIssue({
+      code: 'invalid_format',
+      format: 'regex',
+      message: `Invalid chain alias format: ${value}`,
+    });
+    return;
+  }
+
+  const isKnown = CHAINS.some((chain) => chain.alias === value);
+  if (!isKnown) {
+    ctx.addIssue({
+      code: 'custom',
+      message: `Invalid chain alias: ${value}`,
+    });
+  }
+});
```

Comment:
```suggestion
export const chainAlias = aliasSchema.refine((value) => CHAINS.some((chain) => chain.alias === value), {
  error: (iss) => `Invalid chain alias: ${iss.input}`,
});
```

I'd stick to original implementation in the v4 way.

Reply:
I have tried that at first but tests were failing due to a bug in zod:

```
    expect(received).toThrow(expected)

    - Expected message  - 0
    + Received message  + 1

      [
        {
    +     "origin": "string",
          "code": "invalid_format",
          "format": "regex",
          "pattern": "/^[\\da-z-]+$/",
          "path": [],
          "message": "Invalid string: must match pattern /^[\\da-z-]+$/"
        },
        {
          "code": "custom",
          "path": [],
          "message": "Invalid chain alias: Mantle"
        }
      ]
 ```
 
 I have added `origin` but then I got this error:
```
Object literal may only specify known properties, and 'origin' does not exist in type '$ZodIssueInvalidStringFormat'.ts(2353)
```

Which is true and can be seen [here](https://github.com/colinhacks/zod/blob/592b7b5553b5ef64474ec4ff6d4a9a5fc3c9d409/packages/zod/src/v4/core/errors.ts#L46). But `origin` is being added anyway like following [ZodCheckStringFormat](https://github.com/colinhacks/zod/blob/592b7b5553b5ef64474ec4ff6d4a9a5fc3c9d409/packages/zod/src/v4/core/checks.ts#L808)

There are some issues related to regex on zod repo but I couldn't find any about this specific situation. But you can see the actual behavior is too emit `origin` here is an [issue](https://github.com/colinhacks/zod/issues/4964).